verification/src/solidity/compiler_cache.rs

```diff
@@ -0,0 +1,85 @@
+use ethers_solc::Solc;
+use semver::Version;
+use std::{collections::HashMap, sync::Arc};
+
+#[derive(Default)]
+pub struct CompilerCache {
```

Comment:
May be use `SolcManager` as the name of the structure with `fetch` public method which checks whether the path is in the cache or download it, otherwise? 

The problem with `CompilerCache`, in my opinion, is that when you get something from cache in most cases you expect it to return quickly and definitely not to run a long request downloading compilers from somewhere

Reply:
> is that when you get something from cache in most cases you expect it to return quickly

well that's just not true :) 
it is very common to cache static data for any use, including long term (for example caching popular videos in youtube, which will be watched for a long time)

`ThingManager` imo is not descriptive, how does this struct manages solc? Configures, downloads, stores, loads in the memory, compiles?